From 3d5001bef5a2fe4d330716d6c2bc77a92894ad09 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Mon, 5 Feb 2024 17:48:58 +0100
Subject: [PATCH 507/954] feat(stm32mp21): update revision A ID

On STM32MP21, Revision ID is 0x0000, contrary to generic rule.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4592a7b45a67b8767be73050a9c10eb0ef5215a5 feat(stm32mp21): update revision A ID]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I088612f0fa5f5c727aa1cb708200847d1de0712f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/357550
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 8f193f0d5e..be6d683d20 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -69,7 +69,11 @@
 #define STM32MP257D_PART_NB	U(0xC0002E00)
 #define STM32MP257F_PART_NB	U(0x80002000)
 
+#if STM32MP21
+#define STM32MP2_REV_A		U(0x0000)
+#else /* STM32MP21 */
 #define STM32MP2_REV_A		U(0x1000)
+#endif /* STM32MP21 */
 #define STM32MP2_REV_B		U(0x2000)
 
 /*******************************************************************************
-- 
2.53.0

